pkg/controllers/controlplanemachineset/status.go

```diff
@@ -36,8 +37,25 @@ const (
 // updateControlPlaneMachineSetStatus ensures that the status of the ControlPlaneMachineSet is up to date after
 // the resource has been reconciled.
 func (r *ControlPlaneMachineSetReconciler) updateControlPlaneMachineSetStatus(ctx context.Context, logger logr.Logger, cpms *machinev1.ControlPlaneMachineSet, patchBase client.Patch) error {
-	// TODO: Observe the data from the patchBase.Data() to determine if we actually need to update the status or not.
-	// Then log and update as per the test cases.
+	data, err := patchBase.Data(cpms)
+	if err != nil {
+		return fmt.Errorf("can't fetch patch data from CPMS object: %w", err)
+	}
+
+	// Apply changes only if the patch is not empty
+	if string(data) == "{}" {
+		logger.V(3).Info(notUpdatingStatus)
+
+		return nil
+	}
+
+	err = r.Status().Update(ctx, cpms)
+	if err != nil {
+		return fmt.Errorf("failed to sync status for CPMS object: %w", err)
+	}
```

Comment:
Inline the err declaration please, also change the wording
```suggestion
	if err := r.Status().Update(ctx, cpms); err != nil {
		return fmt.Errorf("failed to sync status for control plane machine set object: %w", err)
	}
```